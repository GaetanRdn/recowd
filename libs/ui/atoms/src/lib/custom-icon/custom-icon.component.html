<object
  [data]="name + '.svg' | safe: 'resourceUrl'"
  height="{{ height }}"
  type="image/svg+xml"
  width="{{ width }}"
>
  <img [src]="name + '.png' | safe: 'url'" alt="{{ name }}" />
</object>
