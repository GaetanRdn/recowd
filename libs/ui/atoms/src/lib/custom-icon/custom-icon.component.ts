import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { Required } from '@recowd/utility-types';
import { SafePipe } from '@recowd/ui/pipes';

@Component({
  selector: 'rc-custom-icon',
  standalone: true,
  imports: [SafePipe],
  templateUrl: './custom-icon.component.html',
  styleUrls: ['./custom-icon.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class CustomIconComponent {
  @Input() @Required() public name!: string;
  @Input() public width = 24;

  private _height?: number;

  get height(): number {
    return this._height ?? this.width;
  }

  @Input() set height(height: number) {
    this._height = height;
  }
}
