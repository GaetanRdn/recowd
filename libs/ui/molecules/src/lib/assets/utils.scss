@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
