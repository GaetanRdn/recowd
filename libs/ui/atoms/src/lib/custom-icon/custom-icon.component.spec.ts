import { TestBed, waitForAsync } from '@angular/core/testing';

import { CustomIconComponent } from './custom-icon.component';
import { Component } from '@angular/core';
import { TemplateLookup } from '@recowd/test/utils';

describe('CustomIconComponent', () => {
  let templateLookup: TemplateLookup<HostComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [HostComponent],
      imports: [CustomIconComponent],
    }).compileComponents();

    templateLookup = new TemplateLookup(HostComponent);
  }));

  it.each`
    name       | width | height
    ${'test'}  | ${32} | ${undefined}
    ${'test2'} | ${16} | ${24}
  `(
    'should create with name $name, width $width and height $height',
    ({ name, width, height }) => {
      templateLookup.hostComponent.name = name;
      templateLookup.hostComponent.width = width;
      templateLookup.hostComponent.height = height;

      templateLookup.detectChanges();

      expect(templateLookup.firstChildElement).toMatchSnapshot();
    }
  );
});

@Component({
  template: ` <rc-custom-icon
    [name]="name"
    [width]="width"
    [height]="height"
  ></rc-custom-icon>`,
})
class HostComponent {
  public name!: CustomIconComponent['name'];
  public width!: CustomIconComponent['width'];
  public height!: CustomIconComponent['height'];
}
