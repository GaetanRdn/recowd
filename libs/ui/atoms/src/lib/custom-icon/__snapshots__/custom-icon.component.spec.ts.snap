// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomIconComponent should create with name test, width 32 and height undefined 1`] = `
<rc-custom-icon>
  <img
    alt="test"
    height="32"
    src="assets/images/test.svg"
    width="32"
  />
</rc-custom-icon>
`;

exports[`CustomIconComponent should create with name test2, width 16 and height 24 1`] = `
<rc-custom-icon>
  <img
    alt="test2"
    height="24"
    src="assets/images/test2.svg"
    width="16"
  />
</rc-custom-icon>
`;
